spec/connection_spec.cr

```diff
@@ -0,0 +1,11 @@
+require "./spec_helper"
+
+describe Avram::Connection do
+  it "displays a helpful error when failing to connect" do
+    conn = Avram::Connection.new("postgres://root:root@localhost:5432/tacoman")
+    message = Regex.new("Failed to connect to databse 'tacoman' with username root.")
+    expect_raises(Avram::ConnectionError, message) do
+      conn.try_connection!
+    end
+  end
```

Comment:
Could you add a spec for one without password to check that it displays the password hint?